src/components/media-controls/media-controls.spec.ts

```diff
@@ -2,40 +2,48 @@ import { expect } from "../../tests/assertions";
 import { mediaControlsFixture as test } from "./media-controls.fixture";
 
 test.describe("audio element communication", () => {
-  test.beforeEach(async ({ fixture }) => {
-    await fixture.create();
-  });
-
-  // this test exists because if you don't correctly import or mount the media controls component
-  // it will still create the element tag, but with no shadow content or error indicating that mounting
-  // the component failed and will make all other tests fail because no content = a "hidden" state
-  // therefore, by having a mounting smoke test, we can ensure that this test will fail only if
-  // we have mounted the component incorrectly
-  test("creating a visible web component", async ({ page }) => {
-    const mediaControls = page.locator("oe-media-controls");
-    await expect(mediaControls).toBeVisible();
-  });
-
-  test("state before interaction", async ({ fixture }) => {
-    const isPlaying = await fixture.isPlayingAudio();
-    expect(isPlaying).toBe(false);
-  });
-
-  test("play functionality", async ({ fixture }) => {
-    await fixture.toggleAudio();
-    const isPlaying = await fixture.isPlayingAudio();
-    expect(isPlaying).toBe(true);
-  });
-
-  test("play pause functionality", async ({ fixture }) => {
-    // start playing audio
-    // by clicking the action button again, we should stop playing audio
-    await fixture.toggleAudio();
-    await fixture.toggleAudio();
-
-    const isPlaying = await fixture.isPlayingAudio();
-
-    expect(isPlaying).toBe(false);
+  [false, true].forEach((useElementId: boolean) => {
```

Comment:
much like your boolean parameter in a method rule, this would be a case where i would split the test apart into different specs for readability, mainly because you're not actually varying a boolean state in the test.

Reply:
true, but there's other ways to refactor tests.

e.g. you could have a common tests function, inside the describe:


```
describe("for with property reference", () => {
  before(() => /*set up*/) 

  commonTests()
})

describe("for with attribute", () => {
  before(() => /*set up*/) 

  commonTests()
})
```

I'm not going to die on this hill, but this is how i'd structure it